Sources/MongoDBVapor/ExtendedJSON+Content.swift

```diff
@@ -0,0 +1,24 @@
+import MongoSwift
+import Vapor
+
+// These extension enable users to encode/decode their Content types to/from JSON using `ExtendedJSONEncoder` and
+// `ExtendedJSONDecoder`. This is useful in cases where users are using one of our custom BSON types in their
+// structs which does not currently support encoding/decoding via `JSONEncoder`/`JSONDecoder`.
+
+extension ExtendedJSONEncoder: ContentEncoder {
+    public func encode<E>(_ encodable: E, to body: inout ByteBuffer, headers: inout HTTPHeaders) throws
+        where E: Encodable
+    {
+        headers.contentType = .json
+        try body.writeBytes(self.encode(encodable))
+    }
+}
+
+extension ExtendedJSONDecoder: ContentDecoder {
+    public func decode<D>(_: D.Type, from body: ByteBuffer, headers _: HTTPHeaders) throws -> D
+        where D: Decodable
+    {
+        let data = body.getData(at: body.readerIndex, length: body.readableBytes) ?? Data()
```

Comment:
For performance reasons I wonder if it would make sense to add `ByteBuffer` APIs to `ExtendedJSONDecoder`. That way we don't need to copy the data here.

Reply:
hmm yeah that might be a good idea. I think right now we end up doing two copies, first this one and then another one is done via `JSONParser` to convert the `Data` to an `[UInt8]`: https://github.com/swift-extras/swift-extras-json/blob/122b9454ef01bf89a4c190b8fd3717ddd0a2fbd0/Sources/ExtrasJSON/Parsing/DocumentReader.swift#L13

we can't get around the second one I guess, but we could avoid the first by accepting a `ByteBuffer`.

I opened [SWIFT-1088](https://jira.mongodb.org/browse/SWIFT-1088)